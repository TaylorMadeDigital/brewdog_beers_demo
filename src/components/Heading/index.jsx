import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import s from './index.scss';

const Heading = ({ children, headingLevel, alignment, addTopPadding }) => {
  const Tag = headingLevel;
  const textAlign = alignment;
  const headingClasses = classNames(s.heading, {
    [s['heading--padding']]: addTopPadding,
  }, s[`heading--${textAlign}`]);
  return (
    <Fragment>
      <Tag data-testid="heading"  className={headingClasses}>
        {children}
      </Tag>
    </Fragment>
  );
}

export const headingPropTypes = {
  headingLevel: PropTypes.string,
  alignment: PropTypes.string,
  addTopPadding: PropTypes.bool,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]),
};

Heading.propTypes = headingPropTypes;

Heading.defaultProps = {
  headingLevel: 'h2',
  alignment: 'center',
  addTopPadding: false,
};

export default Heading;
