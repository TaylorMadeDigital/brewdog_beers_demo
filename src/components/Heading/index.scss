@import '../../assets/scss/grid';
@import '../../assets/scss/typography';
@import '../../assets/scss/variables';
@import '../../assets/scss/mixins';
  
  .heading {
    @include font-size('default', 'bold');
    font-family: $headline-font;
    text-align: center;
    color: $heading-text-color;
    margin: 0;

    &.heading--left {
      text-align: left;
    }
  
    &.heading--padding {
      @include stdSpacing('margin', 'top');
    }
  }
  
  h1.heading {
    color: darken($heading-text-color, 10%);
    margin-bottom: 0;
    @include font-size('x-large', 'bold');
  }
  
  h2.heading {
    margin-bottom: $space-xs;
    @include font-size('large', 'bold');
  }
  
  h3.heading {
    margin-bottom: $space-xs;
    @include font-size('medium', 'bold');
  }
  
  h4.heading {
    margin-bottom: $space-base;
    @include font-size('default', 'bold');
  }
  