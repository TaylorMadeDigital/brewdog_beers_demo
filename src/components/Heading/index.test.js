import React from 'react';
import { render, cleanup } from "@testing-library/react";
import Heading from './index';

describe('<Heading />', () => {

  afterEach(cleanup);

  it('snapshot: renders default state', () => {
    const { asFragment } = render(<Heading>A sample heading</Heading>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('snapshot: renders h4 ', () => {
    const { asFragment } = render(<Heading headingLevel="h4">A sample heading</Heading>);
    expect(asFragment()).toMatchSnapshot();
  });
});
