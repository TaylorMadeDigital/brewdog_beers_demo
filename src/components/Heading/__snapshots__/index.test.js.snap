// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> snapshot: renders default state 1`] = `
<DocumentFragment>
  <h2
    class=""
    data-testid="heading"
  >
    A sample heading
  </h2>
</DocumentFragment>
`;

exports[`<Heading /> snapshot: renders h4  1`] = `
<DocumentFragment>
  <h4
    class=""
    data-testid="heading"
  >
    A sample heading
  </h4>
</DocumentFragment>
`;
